catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -26,157 +33,175 @@
     update(initialValues);
   }
 
-  public void update(Map<String, Object> properties) {
-    if (properties != null && !properties.isEmpty()) {
-      setClassification((String) properties.get(CLASSIFICATION_KEY));
-      setReleasability((String) properties.get(RELEASABILITY_KEY));
-      setCodewords((String) properties.get(CODEWORDS_KEY));
-      setDisseminationControls((String) properties.get(DISSEMINATION_CONTROLS_KEY));
-      setOtherDisseminationControls((String) properties.get(OTHER_DISSEMINATION_CONTROLS_KEY));
-      setOwnerProducer((String) properties.get(OWNER_PRODUCER_KEY));
+  /**
+   * Updates cached default system high attribute values from the {@value
+   * USERS_ATTRIBUTES_FILE_NAME} file. If unable to read the file, none of the cached values will be
+   * updated.
+   */
+  public void update(Map<String, String> properties) {
+    try {
+      cachedMetacardAttributeNameToSystemHighValuesMap = getHighWaterSecurityMarkings(properties);
+      LOGGER.trace("Successfully updated configuration and cached system high attribute values");
+    } catch (DefaultSecurityAttributeValuesPluginException e) {
+      // this should never happen assuming that the configuration includes expected keys, the
+      // expected system properties are set, and the users.attributes file is formatted as expected
+      LOGGER.warn(
+          "Unable to get system high properties. Not updating cached system high attribute values. See the debug log for more details. The Default Security Attribute Plugin may not work as expected. Try restarting the system.");
+      LOGGER.debug("Unable to get system high properties from {}", USERS_ATTRIBUTES_FILE_NAME, e);
+    }
+  }
+
+  private static class DefaultSecurityAttributeValuesPluginException extends Exception {
+
+    DefaultSecurityAttributeValuesPluginException(String message) {
+      super(message);
     }
 
-    this.securityMarkings = getHighwaterSecurityMarkings();
+    DefaultSecurityAttributeValuesPluginException(String message, Exception e) {
+      super(message, e);
+    }
   }
 
   /**
    * Adds default system high-water markings in the event that none of the policies were able to
    * apply security markings to this metacard.
    *
-   * @return Map of the metacard security markings to the value of their corresponding system high
-   *     attribute.
+   * @return metacard with possibly updated markings
    */
-  public Metacard addDefaults(Metacard metacard) {
-
-    if (securityMarkings
-        .keySet()
+  private Metacard addDefaults(final Metacard metacard) {
+    if (CONFIGURATION_KEY_TO_METACARD_ATTRIBUTE_NAME_MAP
+        .values()
         .stream()
         .anyMatch(attributeName -> metacard.getAttribute(attributeName) != null)) {
+      LOGGER.trace(
+          "Assuming that other policies were able to apply security markings. Not adding defaults.");
       return metacard;
     }
 
     if (isRegistryMetacard(metacard)) {
```

Comment:
This isn't part of your changeset but I am surprised to see this here. I think we should be checking if the metacard is a `resource` and if it is not then returning.